### PR TITLE
Spanner connection timeout

### PR DESCRIPTION
**Pull request by charlesliqlogic**
Work towards #3616

### CODE REVIEW

File: google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java [hunk 1]
```diff
@@ -233,13 +236,36 @@ public void cancelled(Context context) {
...
+  static <T> T runWithRetries(Callable<T> callable, RetrySettings retrySettings) {
+    Instant start = Instant.now();
+    Instant end;
+
     // Use same backoff setting as abort, somewhat arbitrarily.
     Span span = tracer.getCurrentSpan();
     ExponentialBackOff backOff = newBackOff();
     Context context = Context.current();
     int attempt = 0;
+    int maxAttempt = 6;
```
Comment (JesseLovelace) [inline google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java]: Can we pull the 6 out into a constant? Same with the 50 below

**Comment by JustinBeckwith** [comment]
@sduskis what is this blocked on?

**Comment by sduskis** [comment]
@JustinBeckwith, I removed the blocked.  @JesseLovelace or @kolea2, who should review this?

**Comment by sduskis** [comment]
I'm going to close this.  We'll need a new PR to fix #3616.

**Pull request closed without merging**